File: app/src/main/java/zmuzik/slidingpuzzle2/common/view/SquareImageView.java
package zmuzik.slidingpuzzle2.common.view;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.ImageView;

public class SquareImageView extends ImageView {

    public SquareImageView(Context context) {
        super(context);
        setScaleType(ScaleType.CENTER_CROP);
    }

    public SquareImageView(Context context, AttributeSet attrs) {
        super(context, attrs);
        setScaleType(ScaleType.CENTER_CROP);
    }

    public SquareImageView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        setScaleType(ScaleType.CENTER_CROP);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        int dim = Math.min(getMeasuredHeight(), getMeasuredWidth());
        setMeasuredDimension(dim, dim);
    }
}

File: app/src/main/java/zmuzik/slidingpuzzle2/common/view/SquareImageView.kt
package zmuzik.slidingpuzzle2.common.view

import android.content.Context
import android.util.AttributeSet
import android.widget.ImageView

class SquareImageView
@JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) :
        ImageView(context, attrs, defStyleAttr) {

    init{
        scaleType = ImageView.ScaleType.CENTER_CROP
    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        val dim = Math.min(measuredHeight, measuredWidth)
        setMeasuredDimension(dim, dim)
    }
}
